### DISCUSSION TITLE
Listary File Dialog Feature Design

### DISCUSSION THREAD

**Discussion by taooceros** [Flow-Launcher/Flow.Launcher#929] [General]
Only plugin registered for path search will be included (A new interface will be created for this)
Name Vote
- [ ] `IPathProvider`
- [ ] `IFileChoser`
- [ ] `IPathChoser`
- [ ] ...

A new type of result that has special file path will be created (or a new field in the current result added)
That path will be injected to the file selector

**Comment by jjw24** [comment]
sorry why is this called Listary File Dialog? is it because it's similar to the Listary function?

If it's for path search, i would vote for IPathProvider

Why do we need a special result, can we not add the new field in the current result model like the autocomplete field?

**Reply by taooceros** [reply]
> sorry why is this called Listary File Dialog? is it because it's similar to the Listary function?

yeah🤣

> Why do we need a special result, can we not add the new field in the current result model like the autocomplete field?

that will be another query model, which only triggered when detecting file dialog up.

**Reply by jjw24** [reply]
so using the existing result model to hold the path will require another query model?

**Comment by AlanEdd** [comment]
This would be a brilliant feature. There are various bits and pieces of software the allow jumping to locations e.g.. Direct Folders, quick Access pop-up, Listary, JumpToFolder (using Everything). It would be nice to use Flow for searching paths in the context of these file  dialogs. Also the Shortcuts plugin could pop up a pinned list and it would switch rather than opening a new explorer etc.

Would be nice additions.

**Reply by jjw24** [reply]
Not sure if you already know, there is a plugin that does one of the features allowing user in the open file dialogue to jump to the path of their windows explorer- DirQuickJump.

**Comment by Arcticn** [comment]
For reference, I found Listary has shared some experience on how to work on it.
https://github.com/listary/Listary.FileAppPlugin/blob/master/docs/Getting%20Started.md
